ZenPacks/zenoss/OpenStackInfrastructure/poll_openstack.py

```diff
@@ -42,7 +45,11 @@ def __init__(self, username, api_key, project_id, auth_url, region_name):
 
     @inlineCallbacks
     def _populateFlavorData(self, nova, data):
-        result = yield nova.flavors(detailed=True, is_public=None)
+        try:
+            result = yield nova.flavors(detailed=True, is_public=None)
+        except Exception:
+            return
```

Comment:
Can this be more selective than Exception?   This swallows any error, which might mask a real problem.. we only need to ignore permission errors here really, if that's possible.   Looks like you did something like that for nova.servers, so it should be straightforward to carry that kind of check across